manifests/server/config.pp

```diff
@@ -29,16 +29,35 @@
     mode   => '0755',
   }
 
-  file { $cfg_file:
-    ensure  => present,
-    owner   => $owner,
-    group   => $group,
-    mode    => '0644',
-    require => [
-      File[$cfg_dir],
-      Class['dns::server::install']
-    ],
-    notify  => Class['dns::server::service'],
+  case $::osfamily {
+    'RedHat': {
+      file { $cfg_file:
+        ensure  => present,
+        owner   => $owner,
+        group   => $group,
+        mode    => '0644',
+        content => template("${module_name}/named.conf.erb"),
```

Comment:
Is there any particular reason why we don't template out this file in both redhat and debian cases (assuming that all the OS specific differences are in the params.pp file?

That would greatly simplify the changes to this file, and hide any OS-specific paths in params.pp